test/test_goodmans_er.py

```diff
@@ -1,6 +1,76 @@
 """Test Goodmans Ecological Regression."""
+import numpy as np
+import pytest
 from pyei.goodmans_er import *  # pylint:disable=wildcard-import,unused-wildcard-import
 
 
-def test_true():
-    assert True
+@pytest.fixture
+def group_and_vote_fractions():
+    """ Sample group and vote fractions, where every member of the demographic
+        group votes for the given candidate and every non-member of the
+        demographic group does not vote for the given candidate.
+    """
+    group_share = np.array([0, 0.2, 0.4, 0.6, 0.8])
+    vote_share = np.array([0, 0.2, 0.4, 0.6, 0.8])
+    return group_share, vote_share
+
+
+@pytest.fixture
+def group_and_vote_fractions_with_pop():
+    """ Sample group and vote fractions, where every member of the demographic
+        group votes for the given candidate and 10% of the demographic group's
+        complement supports the given candidate (i.e., slope = 1, intercept = 0.1),
+        with an exception of one precinct.
+        All precincts have population 1000, except one precinct that has population
+        1 and does not follow the above formula.
+    """
+    group_share = np.array([0, 0.2, 0.4, 0.6, 0.8, 0.9])
+    vote_share = np.array([0.1, 0.3, 0.5, 0.7, 0.9, 0.9])
+    populations = np.array([1000, 1000, 1000, 1000, 1000, 1])
+    return group_share, vote_share, populations
+
+
+def test_fit(group_and_vote_fractions):  # pylint:disable=redefined-outer-name
+    model = GoodmansER()
+    group_share, vote_share = group_and_vote_fractions
+    model.fit(group_share, vote_share)
+    np.testing.assert_almost_equal(model.intercept_, 0)
+    np.testing.assert_almost_equal(model.slope_, 1)
+
+
+def test_weighted_fit(group_and_vote_fractions_with_pop):  # pylint:disable=redefined-outer-name
+    model = GoodmansER(is_weighted_regression=True)
+    group_share, vote_share, pops = group_and_vote_fractions_with_pop
+    model.fit(group_share, vote_share, pops)
+    np.testing.assert_almost_equal(model.intercept_, 0.1, decimal=3)
```

Comment:
this `assert_almost_equal` is because we've got floats, right? or are the coefficients also a little bit off?

Reply:
closer to the latter. the way i'm testing the weighted regression is with 6 "precincts" where 5/6 precincts (which each have population 1,000) follow the equation `vote_share = group_share + 0.1` and the 6th precinct (which has population 1) has `vote_share = group_share`. If the weighted regression works, then the estimated intercept & slope should be very close to 0.1 & 1, respectively, but we expect it to be a _little_ off because of the 1-person precinct. If we weren't doing weighted regression, and each precinct had the same weight, then the estimated intercept & slope would be very different from 0.1 & 1, so this is kind of a check that the weighted regression is giving us a result in line with the intuition that the precinct with very low population should basically be ignored.